lang/python/setup_pip.py

```diff
@@ -285,21 +246,21 @@ def source_filter(sources):
 # Here's the configure/make operations we perform before the python extension
 # is linked.
 configure_cmds = [
-    './makemake --clean-and-make',
-    './reconf',
-    # force building a position independent library; it will be linked
-    # into a single shared library with the SWIG interface code.
-    'CFLAGS="${CFLAGS:-} -fPIC -DPIC" ' + \
-    '../configure --enable-python --with-builtins=' + ','.join(builtin_names)
+    'cmake -B cmake_pip_build -G Ninja -DCMAKE_C_FLAGS="${CFLAGS:-}" -DENABLE_PYTHON=1 ' + \
+    ' '.join(map(lambda name: '-DHAVE_BUILTIN_EXTENSION_' + name.upper() + '=1', builtin_names)),
```

Comment:
I think we need to put a dependency on cmake and ninja to make sure they are installed.  Some info how to do that here: https://setuptools.readthedocs.io/en/latest/userguide/dependency_management.html .  Looks like setting `install_requires` in the `setup()` function call will do the trick.  Would be good to test this on a system that didn't have ninja and/or cmake installed.

Reply:
I tried to add cmake ninja to the python dependency but that is not working. I think the problem is that we install cmake and ninja in python. But when we build we run a shell command in python that doesn't look into the python package so cmake is still not found.

Reply:
I think this is not working because cmake is using a different way to locate external libraries. If I use cmake installed by python, it seems not finding other dependencies even I set LD_LIBRARY_PATH. @alisonfel